tests/e2e/utils/workspace/TestWorkspaceUtil.ts

```diff
@@ -164,31 +164,41 @@ export class TestWorkspaceUtil implements ITestWorkspaceUtil {
         Logger.debug('TestWorkspaceUtil.stopWorkspaceById');
 
         const stopWorkspaceApiUrl: string = `${TestWorkspaceUtil.WORKSPACE_API_URL}/${id}`;
+        let stopWorkspaceResponse;
 
         try {
-            const stopWorkspaceResponse = await this.processRequestHandler.delete(`${stopWorkspaceApiUrl}/runtime`);
-            // response code 204: "No Content" expected
-            if (stopWorkspaceResponse.status !== 204) {
-                throw new Error(`Can not stop workspace. Code: ${stopWorkspaceResponse.status} Data: ${stopWorkspaceResponse.data}`);
-            }
-            let stopped: boolean = false;
-            let wsStatus = await this.processRequestHandler.get(stopWorkspaceApiUrl);
-            for (let i = 0; i < TestConstants.TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS; i++) {
-                wsStatus = await this.processRequestHandler.get(stopWorkspaceApiUrl);
-                if (wsStatus.data.status === 'STOPPED') {
-                    stopped = true;
-                    break;
-                }
-                await this.driverHelper.wait(TestConstants.TS_SELENIUM_DEFAULT_POLLING);
-            }
-            if (!stopped) {
-                let waitTime = TestConstants.TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS * TestConstants.TS_SELENIUM_DEFAULT_POLLING;
-                throw new error.TimeoutError(`The workspace was not stopped in ${waitTime} ms. Currnet status is: ${wsStatus.data.status}`);
-            }
+            stopWorkspaceResponse = await this.processRequestHandler.delete(`${stopWorkspaceApiUrl}/runtime`);
         } catch (err) {
-            console.log(`Stopping workspace failed. URL used: ${stopWorkspaceApiUrl}`);
+            console.log(`Stop workspace call failed. URL used: ${stopWorkspaceApiUrl}/runtime`);
             throw err;
         }
+            
+        // if workspace is already stopped, it will return 409: "Conflict"
+        if (stopWorkspaceResponse.status === 409) {
+            Logger.warn(`TestWorkspaceUtil.stopWorkspaceById Workspace {${id}} is already STOPPED`);
+            return;
+        }
+        
+        // response code 204: "No Content" expected
+        if (stopWorkspaceResponse.status !== 204) {
+            throw new Error(`Can not stop workspace. Code: ${stopWorkspaceResponse.status} Data: ${stopWorkspaceResponse.data}`);
+        }
+
+        let stopped: boolean = false;
+        let wsStatus = await this.processRequestHandler.get(stopWorkspaceApiUrl);
+        for (let i = 0; i < TestConstants.TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS; i++) {
```

Comment:
It's weird to see `TS_SELENIUM_PLUGIN_PRECENCE_ATTEMPTS` here but it doesn't have to be fixed in this PR.

Reply:
yeah, and there's also a typo in the variable name :D
not sure what I should put there instead